### DISCUSSION TITLE
How to set qos on builtin topic

### DISCUSSION THREAD

**Discussion by skysky97** [eProsima/Fast-DDS#2665] [Q&A]
The `get_builtin_subscriber` is not implemented, how can I set qos on builtin reader?

**Comment by JLBuenoLopez** [comment]
Hi @skysky97

Currently it is not possible to modify the QoS of the builtin DataReaders. Why do you want to modify the QoS defined by the stadard? Which QoS do you want to modify? You should be aware that most of the QoS cannot be changed in enabled DataReaders. Currently the only QoS you would be able to modify if you had access to the builtin Subscriber would be [`DeadlineQoSPolicy`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPolicies.html#deadlineqospolicy), [`LifespanQoSPolicy`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPolicies.html#lifespanqospolicy), and [`UserDataQosPolicy`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPolicies.html#userdataqospolicy). The rest of the QoS are immutable and cannot be changed in an enabled DataReader.
